dss/__init__.py

```diff
@@ -34,6 +38,26 @@ def get_logger():
     except RuntimeError:
         return logging.getLogger(__name__)
 
+class OperationWithAuthorizer(Operation):
+    authorized_domains = os.environ["AUTHORIZED_DOMAINS"].split()
+    def oauth2_authorize(self, function):
+        def wrapper(request):
+            if "token_info" in request.context.values:
+                token_info = request.context.values["token_info"]
+                if not int(token_info["expires_in"]) > 0:
+                    raise OAuthProblem(description="Authorization token has expired")
```

Comment:
I'm curious how this interacts with the error reporting code, and if compliance with the swagger spec is enforced.